### PR TITLE
feat: add support for Vite

### PR DESCRIPTION
**Pull request by ascorbic**
## Description

Add Vite plugins for edge and serverless, and demos for both. 

Fixes FRA-267

### CODE REVIEW

File: packages/remix-adapter/src/plugin.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    async configResolved(config) {
+      resolvedConfig = config
+    },
+    async writeBundle() {
+      if (currentCommand === 'build' && isSsr) {
+        // Write the server entrypoint to the Netlify functions directory
+        const functionDir = join(resolvedConfig.root, '.netlify/functions-internal')
+        await mkdir(functionDir, { recursive: true })
+        await writeFile(
+          join(functionDir, 'server.mjs'),
```
Comment (pieh) [inline packages/remix-adapter/src/plugin.ts:70]: There is `server.mjs` which is thin Netlify Function V2 wrapper (with inline config for catch-all) and then `server.js` (which I assume is what vite compilation output entrypoint is). So same base file name with different extension which might make it confusing? Could we maybe name Netlify Function entrypoint a little different (we use `___netlify-server-handler.mjs` in next runtime as example) to differentiate and avoid confusion?

File: packages/remix-adapter/src/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+export default createRequestHandler({ build });
+`
+// This is written to the functions directory. It just re-exports
+// the compiled entrypoint, along with Netlify function config.
+function generateNetlifyFunction(server: string) {
+  return /* js */ `
+    export { default } from "${server}";
+    export const config = {
+      name: "Remix server handler",
+      generator: "${name}@${version}",
```
Comment (pieh) [inline packages/remix-adapter/src/plugin.ts:21]: Is this actually working? `@netlify/functions` types don't mention `name` or `generator` - https://github.com/netlify/functions/blob/cdf899c63519d623e858f4db9d04d55c99456f77/src/function/v2.ts#L7-L43

Comment (pieh) [reply packages/remix-adapter/src/plugin.ts:21]: In Next Runtime we do output .json file with metadata like that currently

Comment (pieh) [reply packages/remix-adapter/src/plugin.ts:21]: I guess same question applies to `excludedPath` in edge plugin

Comment (ascorbic) [reply packages/remix-adapter/src/plugin.ts:21]: I'm basing it on this: https://docs.netlify.com/edge-functions/create-integration/#generate-declarations

Comment (ascorbic) [reply packages/remix-adapter/src/plugin.ts:21]: ...and https://docs.netlify.com/edge-functions/declarations/#declare-edge-functions-inline

**Comment by ascorbic** [comment]
@pieh I'll do the smoke test update in a follow-up, because I don't want to add the demo sites until this is merged to main

File: packages/remix-adapter/src/plugin.ts (hunk 2 above)
Comment (pieh) [reply packages/remix-adapter/src/plugin.ts:21]: Didn't test edge functions - but lambda ones don't work without `.json` with `generator` and `name`

Comment (pieh) [reply packages/remix-adapter/src/plugin.ts:21]: as in - they work - just don't fields don't do anything - I needed `.json` file to have `displayName` + `generator` show up

**Review by pieh — APPROVED** [review]
I'll approve as-is to allow move this forward - the `name`/`generator` metadata for lambda plugin as-is doesn't seem to do anything but also not breaking so this can be handled in follow possibly? Up to you

File: packages/remix-adapter/src/plugin.ts (hunk 2 above)
Comment (pieh) [reply packages/remix-adapter/src/plugin.ts:21]: You can try `ntl build` locally and check `.netlify/functions/manifest.json`

For this:
```
export const config = {
  path: "/no-meta",
  name: "Test",
  generator: "custom",
};
```

I get in manifest
```
{
      "generator": "internalFunc",
}
```

and for .json meta I get
```
{
      "displayName": "Custom name",
      "generator": "@pieh/wat@0.0.0-alpha.0",
}

**Comment by ascorbic** [comment]
Thanks for digging into this. It seems like somethign that should work, so I'll either follow up with the manifest or see if compute plan to fix it

**Pull request merged**